code/modules/mob/living/carbon/examine.dm

```diff
@@ -161,6 +162,32 @@
 			// add any extra info on the limbs themselves
 			msg += examine_handle_individual_limb(limb_name)
 
+	// hallucinating?
+	if(hallucinating && prob(25))
+		// List of hallucination messages
+		var/list/hallucination_texts = list(
+			"You blink, and for a moment, [p_their()] body shimmers like a mirage, [p_their()] gaze unsettlingly intense.",
+			"[p_they(TRUE)] appear[p_s()] to be surrounded by a swarm of tiny, glowing butterflies.",
+			"[p_they(TRUE)] [p_are()] wearing a crown made of spaghetti. Wait, no... it's gone now.",
+			"[p_they(TRUE)] look[p_s()] suspicious, as if plotting a jelly heist.",
+			"[p_they(TRUE)] begin[p_s()] to hum a tune, but the sound seems to echo from all directions at once.",
+			"[p_they(TRUE)] smile [p_s()], and for a second, [p_their()] face twists into a thousand tiny reflections.",
+			"[p_they(TRUE)] seem[p_s()] to float slightly above the ground, [p_their()] feet just brushing against the floor.",
+			"[p_their(TRUE)] hands flicker like holograms, shifting between different gestures before returning to normal.",
+			"[p_they(TRUE)] seems to be cloaked in a faint, swirling fog that disappears the moment you focus on it.",
+			"You glance at [p_them()], and for an instant, [p_their()] shadow stretches unnaturally long, as if reaching for something just out of view. Did that shadow have a face?",
+			"You glance at [p_them()], and for a moment, [p_their()] eyes seem to flash with a strange, metallic gleam. You could have sworn it was gold... or was it red?",
+			"[p_they(TRUE)] walk[p_s()] past, [p_their()] silhouette stretching longer than it should. Were [p_their()] footsteps too quiet? Or is it just you? There's something off about the way [p_they()] move[p_s()].",
```

Comment:
This one might read weirdly if the person is unconscious or prone or otherwise immobile. To me it reads as asserting they ARE walking, when they might not be. I want to suggest changing "they walk past" to "they walk straight towards you" or "they walk away from you" to make the walking itself read as part of the hallucination. 
Disregard if you think "they walk past" will read as part of the hallucination rather than an assertion that they're actually walking past.

Reply:
updated it! 😄